bindings/pyroot/cppyy/CPyCppyy/src/CPPExcInstance.cxx

```diff
@@ -150,6 +150,77 @@ static PyObject* ep_richcompare(CPPExcInstance* self, PyObject* other, int op)
     return PyObject_RichCompare(self->fCppInstance, other, op);
 }
 
+//----------------------------------------------------------------------------
+static int ep_nonzero(CPPExcInstance* self)
+{
+   if (self->fCppInstance)
+      return PyObject_IsTrue(self->fCppInstance);
+
+   return 0;
+}
+
+//-----------------------------------------------------------------------------
+static PyNumberMethods ep_as_number = {
+    0,                             // nb_add
+    0,                             // nb_subtract
+    0,                             // nb_multiply
+#if PY_VERSION_HEX < 0x03000000
+    0,                             // nb_divide
+#endif
+    0,                             // nb_remainder
+    0,                             // nb_divmod
+    0,                             // nb_power
+    0,                             // nb_negative
+    0,                             // nb_positive
+    0,                             // nb_absolute
+    (inquiry)ep_nonzero,           // nb_bool (nb_nonzero in p2)
+    0,                             // nb_invert
+    0,                             // nb_lshift
+    0,                             // nb_rshift
+    0,                             // nb_and
+    0,                             // nb_xor
+    0,                             // nb_or
+#if PY_VERSION_HEX < 0x03000000
+    0,                             // nb_coerce
+#endif
+    0,                             // nb_int
+    0,                             // nb_long (nb_reserved in p3)
+    0,                             // nb_float
+#if PY_VERSION_HEX < 0x03000000
+    0,                             // nb_oct
+    0,                             // nb_hex
+#endif
+    0,                             // nb_inplace_add
+    0,                             // nb_inplace_subtract
+    0,                             // nb_inplace_multiply
+#if PY_VERSION_HEX < 0x03000000
+    0,                             // nb_inplace_divide
+#endif
+    0,                             // nb_inplace_remainder
+    0,                             // nb_inplace_power
+    0,                             // nb_inplace_lshift
+    0,                             // nb_inplace_rshift
+    0,                             // nb_inplace_and
+    0,                             // nb_inplace_xor
+    0                              // nb_inplace_or
+#if PY_VERSION_HEX >= 0x02020000
+    , 0                            // nb_floor_divide
+#if PY_VERSION_HEX < 0x03000000
+    , 0                            // nb_true_divide
+#else
+    , 0                            // nb_true_divide
+#endif
```

Comment:
Won't this always result in exactly the same `<whatever the type is>`, because both branches are the same.

Reply:
Nice catch! Although since it is harmless (and it is what is in upstream cppyy now) I will just leave it as it is.

Reply:
Or rather, in the case of the code where the above was copied from, it did matter:
```
#if PY_VERSION_HEX < 0x03000000
    , 0                            // nb_true_divide
#else
    , (binaryfunc)op_div_stub      // nb_true_divide
#endif
```
since for p2, op_div_stub is slotted into nb_divide instead of nb_true_divide. It only looks like it doesn't matter for `CPPExcInstance` as contrary to `CPPInstance` it does not have a divide stub.